PropertyInformationApi/V1/UseCase/GetPropertiesUseCase.cs

```diff
@@ -26,6 +28,10 @@ public IList<HousingProperty> Execute(GetPropertiesRequest request)
                 throw new InvalidQueryParameterException("The postcode parameter does not have a valid format");
 
             var response = _gateway.GetPropertiesByPostcodeOrAddress(request);
+
+            if (!response.Any())
+                throw new PropertyNotFoundException();
```

Comment:
Is this a Hackney convention?  It can be argued either way, 
https://stackoverflow.com/questions/26845631/is-it-correct-to-return-404-when-a-rest-resource-is-not-found
https://softwareengineering.stackexchange.com/questions/203492/when-to-use-http-status-code-404-in-an-api

I think a 404 with a detailed error message in the response is just as useful. Returning a 200 with an empty array is equivalent to returning a 200 and null.

> Of course, it would be good to have this logic in the gateway, use case and controller tested, like for instance testing a controller's 404 catch block:
> 
> https://github.com/LBHackney-IT/training-project-arrears-action-api/blob/d9eb321280abe890837e41ba14ae70da31ee476e/ArrearsActionAPI.UnitTests/V1/Controllers/ArrearsActionControllerTests.cs#L206-L224
> 
> But I assume the work is urgent.

This API doesn't have any tests!